sahi/model.py

```diff
@@ -663,3 +666,161 @@ def _create_object_prediction_list_from_original_predictions(
+        Args:
+            image: np.ndarray
+                A numpy array that contains the image to be predicted. 3 channel image should be in RGB order.
+            image_size: int
+                Inference input size.
+        """
+        if self.model is None:
+            raise ValueError("model not loaded.")
+
+        from sahi.utils.torchvision import numpy_to_torch, resize_image
+
+        if self.image_size is not None:
+            image = resize_image(image, self.image_size)
+            image = numpy_to_torch(image)
+            prediction_result = self.model([image])
+
+        else:
+            prediction_result = self.model([image])
+
+        self._original_predictions = prediction_result
+
+    @property
+    def num_categories(self):
+        """
+        Returns number of categories
+        """
+        return len(self.category_mapping)
+
+    @property
+    def has_mask(self):
+        """
+        Returns if model output contains segmentation mask
+        """
+        return self.model.with_mask
+
+    @property
+    def category_names(self):
+        return self.category_mapping
+
+    def _create_object_prediction_list_from_original_predictions(
+        self,
+        shift_amount_list: Optional[List[List[int]]] = [[0, 0]],
+        full_shape_list: Optional[List[List[int]]] = None,
+    ):
+        """
+        self._original_predictions is converted to a list of prediction.ObjectPrediction and set to
+        self._object_prediction_list_per_image.
+        Args:
+            shift_amount_list: list of list
+                To shift the box and mask predictions from sliced image to full sized image, should
+                be in the form of List[[shift_x, shift_y],[shift_x, shift_y],...]
+            full_shape_list: list of list
+                Size of the full image after shifting, should be in the form of
+                List[[height, width],[height, width],...]
+        """
+        original_predictions = self._original_predictions
+        category_mapping = self.category_mapping
+
+        # compatilibty for sahi v0.8.20
+        if isinstance(shift_amount_list[0], int):
+            shift_amount_list = [shift_amount_list]
+        if full_shape_list is not None and isinstance(full_shape_list[0], int):
+            full_shape_list = [full_shape_list]
+
+        # parse boxes, masks, scores, category_ids from predictions
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        prediction_class = [COCO_CLASSES[i] for i in list(original_predictions[0]["labels"].numpy())]
+        prediction_boxes = [
+            [(i[0], i[1]), (i[2], i[3])] for i in list(original_predictions[0]["boxes"].detach().numpy())
+        ]
+        prediction_score = list(original_predictions[0]["scores"].detach().numpy())
+        prediction_thresh = [prediction_score.index(x) for x in prediction_score if x > self.confidence_threshold][-1]
```

Comment:
@kadirnar bu dogru olmamis. 801. satirda baslayan for loopu icinde kontrol etmelisin confidence scorelari. 312.stari ornek alabilirsin.

Reply:
@kadirnar buna bakma firsatin oldu mu?